ochazuke/webhooks/helpers.py

```diff
@@ -0,0 +1,235 @@
+        result |= a ^ b
+    return result == 0
+
+
+def is_github_hook(request):
+    """Validate the github webhook HTTP POST request."""
+    if request.headers.get('X-GitHub-Event') is None:
+        return False
+    post_signature = request.headers.get('X-Hub-Signature')
+    if post_signature:
+        # TODO: DETERMINE WHERE TO HIDE SECRET KEY
+        key = app.config['HOOK_SECRET_KEY']
+        return signature_check(key, post_signature, request.data)
+    return False
+
+
+def process_issue_event_info(payload):
+    """Extract information we need when handling webhook for issue events."""
+    # Extract the event-specific info and make our own timestamp
+    event_timestmp = datetime.utcnow().replace(microsecond=0).isoformat() + 'Z'
+    action = payload.get('action')
+    milestone = payload.get('issue')['milestone']
+    # If there is no milestone data, let title be None
+    if milestone is not None:
+        milestone_title = milestone.get('title')
+    else:
+        milestone_title = None
+    # Let details be None for opening/closing, but preserve old title on edits
+    if action in ['opened', 'closed', 'reopened', 'edited']:
+        changes = payload.get('changes')
+        if changes is not None:
+            details = {'old title': changes['title']['from']}
+        else:
+            details = None
+    elif action == ('milestoned' or 'demilestoned'):
+        details = {'milestone title': payload.get(
+                    'issue')['milestone']['title']}
+    else:
+        details = {'label name': payload.get('label')['name']}
+    # Create a concise issue event dictionary
+    issue_event_info = {'issue_id': payload.get('issue')['number'],
+                        'header': payload.get('issue')['title'],
+                        'created_at': payload.get('issue')['created_at'],
+                        'milestone': milestone_title,
+                        'actor': payload.get('sender')['login'],
+                        'action': action,
+                        'details': details,
+                        'received_at': event_timestmp}
+    # Pass dictionary to appropriate processor depending on event type
+    if action == 'opened':
+        new_issue(issue_event_info)
+    if action == 'edited':
+        issue_header_edit(issue_event_info)
+    if action == ('closed' or 'reopened'):
+        issue_status_change(issue_event_info)
+    if action == ('milestoned' or 'unmilestoned'):
+        issue_milestone_change(issue_event_info)
+    if action == ('labeled' or 'unlabeled'):
+        issue_label_change(issue_event_info)
+    # Store all new events except assigned/unassigned (filtered out in route)
+    new_event(issue_event_info)
+
+
+def new_issue(info):
+    """Create an issue object to insert into db from an 'opened' issue event.
+
+    When a new issue is opened, we insert it into our issue table, including:
+    - github number (int, 'id')
+    - title (text, 'header')
```

Comment:
What's the reasoning for using `header` to describe the title? Is title being used somewhere else?

Reply:
@miketaylr I had this idea that it would be helpful not to have `label.title`, `milestone.title`, and `issue.title`, because I thought having common property names might get confusing, but that may have just made things... more confusing? 🤷‍♀️ Not at all wedded to it if it's a head-scratcher for everyone else. Github decided that labels have names and milestones and issues have titles (for whatever reason). We could change to follow that if it makes more sense.

Reply:
Gotcha... I'm not 100% against using `header`, if we add a comment somewhere explaining that a header is an abstraction over a `name` and a `title`, depending on the object. 

Otherwise future me will be confused when I go to the GItHub API docs and find no mentions of headers. :)~